src/blr/cli/filterclusters.py

```diff
@@ -1,367 +1,306 @@
+                                            min_reads=args.threshold, summary=summary)
+
+        summary.tot_reads = infile.mapped + infile.unmapped
+        summary.unmapped_reads = infile.unmapped
+        summary.mapped_reads = infile.mapped
+        summary.non_analyzed_reads = summary.unmapped_reads + summary.non_tagged_reads + summary.overlapping_reads_in_pb
+
+    # Writes filtered out
+    with pysam.AlignmentFile(args.x2_bam, "rb") as openin:
+        with pysam.AlignmentFile(args.output, "wb", template=openin) as openout:
+            logger.info("Writing filtered bam file")
+            for read in tqdm(openin.fetch(until_eof=True)):
+                barcode = fetch_bc(pysam_read=read, barcode_tag=args.barcode_tag)
+
+                # If barcode is not in allMolecules the barcode does not have enough proximal reads to make a single
+                # molecule. If the barcode has more than <Max_molecules> molecules, remove it from the read.
+                if barcode in molecule_dict and len(molecule_dict[barcode]) > args.Max_molecules:
+                    read = strip_barcode(pysam_read=read ,barcode_tag=args.barcode_tag)
+
+                    summary.reads_with_removed_barcode += 1
+                    if not barcode in summary.removed_barcodes:
+                        summary.removed_barcodes.add(barcode)
+                        summary.removed_molecules += len(molecule_dict[barcode])
+
+                openout.write(read)
+
+    summary.printStats(barcode_tag=args.barcode_tag, molecule_dict=molecule_dict)
 
-    # Open file, loop over all reads
-    logger.info(f'Running analysis with {"{:,}".format(args.window)} bp window size')
-    logger.info('Fetching reads')
-    progress = BLR.ProgressReporter('Reads processed', 1000000)
-    with pysam.AlignmentFile(args.x2_bam, 'rb') as infile:
-        for read in infile.fetch(until_eof=True):
+    # Write molecule/barcode file stats
+    if args.stats_file:
+        logger.info("Writing statistics files")
+        summary.writeMoleculeStats(output_prefix=args.stats_file, molecule_dict=molecule_dict)
 
-            # Progress reporting. Upmost because of continue-statement in loop
-            progress.update()
+def build_molecule_dict(pysam_openfile, barcode_tag, window, min_reads, summary):
+    """
+    Build allMolecules.final_dict, [barcode][moleculeID] = molecule where molecule are instances of Molecule object,
+    defined as more than <min_reads> within <window> from each other.
+    :param pysam_openfile: Pysam open file instance.
+    :param barcode_tag: Tag used to store barcode in bam file (usually BC).
+    :param window: Max distance between reads to include in the same molecule.
+    :param min_reads: Minimum reads to include molecule in allMolecules.final_dict
+    :param summary: Custom summary intsance
+    :return: dict[barcode][molecule] = moleculeInstance
+    """
+
+    allMolecules = AllMolecules(min_reads=min_reads)
```

Comment:
So as i understand PEP8 (You wanted comments on this ;) ) this naming of instance variables with uppercase should be avoided. I see this also for the class methods which should also be all lowercase in my understanding of the quidelines. Options in this case are: `allmolecules` or `all_molecules`, whichever is more clear. This could however be part of a later pull request.

Reply:
I agree, however ```allMolecules``` and ```AllMolecules``` are an object (sometimes referred to as class) and an instance of that class which should be . Instance variables are those defined as ```self.<variable_name>```, which I believe all are named in lower case and separated by ```_```. 

However, I believe class methods are underlying functions to these classes, which currently has a similar naming system which should be changed (e.g ```addRead``` should be ```addread```).

I'm unable to find anything on naming conventions of the instances (the ```allMolecules```) so I thought it best to leave it for now.

Reply:
I was using the wrong wording with 'instance varliables' but I think that I am right in that `allMolecules` is a variable that stores an instance of the `AllMolecules`. As such it is variable and should follow those conventions. Am I making sense??

Reply:
Yeah that sounds about right. I also found one reference to mixedCapitalization which is supposed to be avoided at all costs, except for backwards compatibility so it's probably best to avoid it.